src/main/java/gg/solarmc/clans/command/commands/clans/InfoCommand.java

```diff
@@ -73,19 +73,24 @@ private void sendClanInfo(CommandSender player, Clan clan, PluginHelper helper)
         int assists = clan.currentAssists();
         int deaths = clan.currentDeaths();
         StringBuilder members = new StringBuilder();
-        clan.currentMembers().forEach(it ->
-                getPlayerNameById(player.getServer(), it.userId())
-                        .thenAccept(members::append));
 
-        Component info = helper.replaceText(plugin.getPluginConfig().clan().info(),
-                Map.of("{clan}", clanName,
-                        "{kills}", String.valueOf(kills),
-                        "{assists}", String.valueOf(assists),
-                        "{deaths}", String.valueOf(deaths),
-                        "{ally}", allyClanName,
-                        "{members}", members.toString()));
 
-        player.sendMessage(info);
+        CentralisedFuture[] memberFutures = (CentralisedFuture[]) clan.currentMembers().stream().map(it ->
+                getPlayerNameById(player.getServer(), it.userId())
+                        .thenAccept(a -> members.append(a).append(",")))
+                .toArray();
+
+        helper.futuresFactory(player.getServer()).allOf(memberFutures).thenRunSync(() -> {
+            Component info = helper.replaceText(plugin.getPluginConfig().clan().info(),
+                    Map.of("{clan}", clanName,
+                            "{kills}", String.valueOf(kills),
+                            "{assists}", String.valueOf(assists),
+                            "{deaths}", String.valueOf(deaths),
+                            "{ally}", allyClanName,
+                            "{members}", members.toString()));
+
+            player.sendMessage(info);
+        });
```

Comment:
You need an exception handler on `thenRunSync`.

On every trailing future - a future which is not returned and has no dependent operation - there needs to be an exception handler.